src/transactions/transaction.js

```diff
@@ -33,19 +34,26 @@ import { prepareTransaction } from './utils';
  * @return {Object}
  */
 
+
 function createTransaction(recipientId, amount, secret, secondSecret, data, timeOffset) {
 	const keys = crypto.getKeys(secret);
+	const fee = data ? (constants.fees.send + constants.fees.data) : constants.fees.send;
 	const transaction = {
 		type: 0,
 		amount,
-		fee: constants.fees.send,
+		fee,
 		recipientId,
 		senderPublicKey: keys.publicKey,
 		timestamp: slots.getTimeWithOffset(timeOffset),
 		asset: {},
-		data,
 	};
 
+	if (data && data.length > 0) {
+		const encoding = jschardet.detect(data).encoding;
+		if (encoding !== 'ascii' && encoding !== 'UTF-8') throw new Error(`invalid encoding ${encoding} in transaction data`);
```

Comment:
(`'utf8'` is the default encoding, but we could specify it to make it explicit).